### PR TITLE
[DOCS] Adds Configuration section to the client docs

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds a `Configuration` section to the .NET docs with some configuration options.

### CODE REVIEW

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -0,0 +1,252 @@
+[[configuration]]
+== Configuration
+
+Connecting to {es} with the client is easy, but it's possible that you'd like to 
+change the default connection behaviour. There are a number of configuration 
+options available on `ConnectionSettings` that can be used to control how the 
+client interact with {es}.
+
+=== Options on ConnectionConfiguration
```
Comment (stevejgordon) [inline docs/configuration.asciidoc:9]:
```suggestion
=== Options on ElasticsearchClientSettings
```

File: docs/configuration.asciidoc [hunk 2]
```diff
@@ -0,0 +1,252 @@
+[[configuration]]
+== Configuration
+
+Connecting to {es} with the client is easy, but it's possible that you'd like to 
+change the default connection behaviour. There are a number of configuration 
+options available on `ConnectionSettings` that can be used to control how the
```
Comment (stevejgordon) [inline docs/configuration.asciidoc:6]:
```suggestion
options available on `ElasticsearchClientSettings` that can be used to control how the 
```

File: docs/configuration.asciidoc [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+Connecting to {es} with the client is easy, but it's possible that you'd like to 
+change the default connection behaviour. There are a number of configuration 
+options available on `ConnectionSettings` that can be used to control how the 
+client interact with {es}.
+
+=== Options on ConnectionConfiguration
+
+The following is a list of available connection configuration options on 
+`ConnectionConfiguration`:
```
Comment (stevejgordon) [inline docs/configuration.asciidoc:12]:
```suggestion
`ElasticsearchClientSettings`:
```

File: docs/configuration.asciidoc [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
+Whether the request should be sent with chunked Transfer-Encoding.
+
+`UserAgent`::
+
+The user agent string to send with requests. Useful for debugging purposes to 
+understand client and framework versions that initiate requests to {es}.
+
+
+==== Options on ConnectionSettings
```
Comment (stevejgordon) [inline docs/configuration.asciidoc:203]: All options are now on the `ElasticsearchClientSettings` so these can be combined into the above list.

File: docs/configuration.asciidoc [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+var connectionSettings = new ConnectionSettings()
+    .DefaultMappingFor<Project>(i => i
+        .IndexName("my-projects")
+        .IdProperty(p => p.Name)
+    )
+    .EnableDebugMode()
+    .PrettyJson()
+    .RequestTimeout(TimeSpan.FromMinutes(2));
+
+var client = new ElasticClient(connectionSettings);
```
Comment (stevejgordon) [inline docs/configuration.asciidoc:251]:
```suggestion
var settings= new ElasticsearchClientSettings()
    .DefaultMappingFor<Project>(i => i
        .IndexName("my-projects")
        .IdProperty(p => p.Name)
    )
    .EnableDebugMode()
    .PrettyJson()
    .RequestTimeout(TimeSpan.FromMinutes(2));

var client = new ElasticsearchClient(settings);
```

File: docs/configuration.asciidoc [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+specified and no default indices are specified for the given CLR type specified 
+for the request.
+
+`DefaultMappingFor`::
+
+Specify how the mapping is inferred for a given CLR type. The mapping can infer 
+the index, id and relation name for a given CLR type, as well as control 
+serialization behaviour for CLR properties.
+
+==== ConnectionSettings with ElasticClient
```
Comment (stevejgordon) [inline docs/configuration.asciidoc:236]:
```suggestion
==== ElasticsearchClientSettings with ElasticsearchClient
```

**Review by stevejgordon — CHANGES_REQUESTED** [review]
LGTM with some changes to align with the new type names in v8.

I'll need to revisit this list after v8 is closer to GA to add/remove any settings as necessary. But let's merge what we have for now.

File: docs/configuration.asciidoc (hunk 4 above)
Comment (szabosteve) [reply docs/configuration.asciidoc]: I combined the two lists of settings into one via [2c434a9](https://github.com/elastic/elasticsearch-net/pull/6387/commits/2c434a9b52306b04cbc298496a6c858321b6f3a8).

**Review by stevejgordon — APPROVED** [review]
LGTM!

**Comment by stevejgordon** [comment]
Thanks @szabosteve!

**Pull request merged**